tests/pages/nodes/NodesPage-cy.js

```diff
@@ -1,97 +1,117 @@
 describe('Nodes Page', function () {
 
-  beforeEach(function () {
-    cy.configureCluster({
-      mesos: '1-for-each-health',
-      nodeHealth: true
-    });
-    cy.visitUrl({url: '/nodes'});
-  });
-
   context('Filters nodes table', function () {
 
+    beforeEach(function () {
+      cy.configureCluster({
+        mesos: '1-for-each-health',
+        nodeHealth: true
+      });
+      cy.visitUrl({url: '/nodes'});
+      cy.get('tbody tr td:first-child').as('hostnames');
+    });
+
     it('should show all nodes', function () {
-      cy.get('tbody tr').should('contain', 'dcos-01');
-      cy.get('tbody tr').should('contain', '167.114.218.155');
-      cy.get('tbody tr').should('contain', '167.114.218.156');
+      cy.get('@hostnames')
+        .should('contain', 'dcos-01')
+        .should('contain', '167.114.218.155')
+        .should('contain', '167.114.218.156');
     });
 
     it('should only show healthy node', function () {
       cy.get('.filter-bar').as('filterBar');
       cy.get('@filterBar').contains('Healthy').click();
 
-      cy.get('tbody tr').should('not.contain', 'dcos-01');
-      cy.get('tbody tr').should('not.contain', '167.114.218.156');
-      cy.get('tbody tr').should('contain', '167.114.218.155');
+      cy.get('@hostnames')
+        .should('not.contain', 'dcos-01')
+        .should('not.contain', '167.114.218.156')
+        .should('contain', '167.114.218.155');
     });
 
     it('should only show unhealthy node', function () {
       cy.get('.filter-bar').as('filterBar');
       cy.get('@filterBar').contains('Unhealthy').click();
 
-      cy.get('tbody tr').should('not.contain', 'dcos-01');
-      cy.get('tbody tr').should('not.contain', '167.114.218.155');
-      cy.get('tbody tr').should('contain', '167.114.218.156');
+      cy.get('@hostnames')
+        .should('not.contain', 'dcos-01')
+        .should('not.contain', '167.114.218.155')
+        .should('contain', '167.114.218.156');
     });
 
     it('should only show nodes with service', function () {
       cy.get('.filter-bar').as('filterBar');
       cy.get('@filterBar').contains('Filter by Service').click();
       cy.get('.dropdown-menu').contains('cassandra-healthy').click();
 
-      cy.get('tbody tr').should('not.contain', '167.114.218.156');
-      cy.get('tbody tr').should('not.contain', '167.114.218.155');
-      cy.get('tbody tr').should('contain', 'dcos-01');
+      cy.get('@hostnames')
+        .should('not.contain', '167.114.218.156')
+        .should('not.contain', '167.114.218.155')
+        .should('contain', 'dcos-01');
     });
 
   });
 
-  context('Filters nodes grid', function () {
+  context('Nodes grid', function () {
 
     beforeEach(function () {
-      cy.visitUrl({url: '/nodes/grid'});
+      cy.configureCluster({
+        mesos: '1-for-each-health',
+        nodeHealth: true
+      });
```

Comment:
Adding this down here now, because we moved the `beforeEach` from the code above into the context above.